core/src/main/java/org/jdbi/v3/core/array/SqlArrayArgumentFactory.java

```diff
@@ -41,27 +39,24 @@
 public class SqlArrayArgumentFactory implements ArgumentFactory {
     @Override
     public Optional<Argument> build(Type type, Object value, ConfigRegistry config) {
-        Class<?> erasedType = GenericTypes.getErasedType(type);
+        Class<?> rawClass = GenericTypes.getErasedType(type);
 
-        if (!(erasedType.isArray() || Collection.class.isAssignableFrom(erasedType))) {
+        if (!rawClass.isArray() && !Collection.class.isAssignableFrom(rawClass)) {
             return Optional.empty();
         }
 
         if (value == null) {
             return Optional.of(new NullArgument(Types.ARRAY));
         }
 
-        Function<Type, Optional<SqlArrayType<?>>> lookup =
-                eT -> config.get(SqlArrayTypes.class).findFor(eT);
-
-        if (erasedType.isArray()) {
-            Class<?> elementType = erasedType.getComponentType();
-            return lookup.apply(elementType)
-                    .map(arrayType -> new SqlArrayArgument<>(arrayType, value));
-        }
-
-        return GenericTypes.findGenericParameter(type, Collection.class)
-                .flatMap(lookup)
+        if (rawClass.isArray()) {
+            return config.get(SqlArrayTypes.class)
+                .findFor(rawClass.getComponentType())
```

Comment:
It still irks me slightly that the `findFor` bit is duplicated here.  I'll make one quick fixup then it's good to go.